modules/util/portal-tools-sample-sql-builder/src/main/java/com/liferay/portal/tools/sample/sql/builder/DataFactory.java

```diff
@@ -5639,6 +5639,27 @@ public List<SegmentsEntryModel> newSegmentsEntryModels(long groupId) {
 		return segmentsEntryModels;
 	}
 
+	public SegmentsExperienceModel newSegmentsExperienceModel(
+		List<LayoutModel> layoutModels) {
+
+		long groupId = 0;
+		long plid = 0;
+
+		for (LayoutModel layoutModel : layoutModels) {
+			long classNameId = layoutModel.getClassNameId();
+
+			if (classNameId != 0) {
+				groupId = layoutModel.getGroupId();
+				plid = layoutModel.getClassPK();
```

Comment:
@slnn Currently the list has 2 `LayoutModel`s, one with `classNameId = 0`, the other with non-0 `classNameId`, and the second one's `classPK` is the `plid` of the first one. So where don't we just do
```
if (classNameId == 0) {
    groupId = layoutModel.getGroupId();
    plid = layoutModel.getPlid();
```

Reply:
Hi @dantewang 

Yes, we can do it, so we prefer to use if(classNameId == 0) instead of if (classNameId != 0) or use plid = layoutModel.getPlid(); can show the relationship more clear?

Reply:
@slnn Yes. I needed to check the code that generated that list to understand why you use `getClassPK` not `getPlid()` with your original implementation.

Reply:
Done